doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,1151 @@
+
+- A New OrchAgent module, ArsOrch, is introduced to orchestrate ARS-related functionalities such as,
+     - Support for ARS profile 
+     - Support for ARS object
+     - Enable/Disable interfaces for ARS
+     - Support ARS nexthops config for ARS object selection
+     - Support ARS over portchannel
+     - Enable/Disable portchannels for ARS object selection
+     - Allow configuration of path metrics
+
+- ACL changes<br>
+When a new ACL table is created, SAI needs to receive a list of supported actions that can be allowed to rules within that table.
+To enable support for the new ARS disable action, the custom table type schema will be extended to include a new action attribute 
+- "DISABLE_ARS_FORWARDING" as part of the actions attribute field.
+
+**Table interactions**:
+
+1. ARS_PROFILE defines the ARS global configuration parameters. ArsOrch uses this to create the ARS profile in SAI. 
+   Currently, only a single profile is supported. If multiple profiles are required, the SAI implementation must also be enhanced to accommodate this functionality.
+    
+2. ARS_OBJECT table used to specify ARS object configuration parameters. ArsOrch uses this to create the ARS SAI object.
+
+3. ARS_INTERFACES holds list of L2 egress interfaces which are ARS-enabled. NHG will be considered ARS-enabled when all of its associated nexthop interfaces are present in the configured ARS_INTERFACES when the match mode configured in ARS_PROFILE is interface-based.
+
+4. ARS_NEXTHOPS holds list of nexthops which are used for identifying ARS-enabled NHG. This table is applicable only when ars_nhg_path_selector_mode in ARS_PROFILE is nexthop.
+NHG will be considered ARS-enabled when all of its associated nexthops are present in the configured ARS_NEXTHOPS entries and additionally its interface should be L2 ARS-enabled interfaces.
+
+5. ARS_PORTCHANNELS table hold list of lag interfaces enabled for ARS.
+
+There are different options for NHG creation. Two modes of operation are supported:
+
+1. NHG created by RouteOrch. APPL_DB adds entry to ROUTE_TABLE with explicit set of nexthops.
+
+2. NHG created by NhgOrch. APPL_DB first adds an entry to the NEXTHOP_GROUP_TABLE and later adds an entry to the ROUTE_TABLE with a reference to the previously created NHG. This mode of operation will be supported only when the Nexthop group's SAI ARS object id attribute "set" capability is supported, to prevent NHG recreations when multiple routes point to the updated NHG.
+
+**ARS selection for Nexthop group**
+
+The ars_nhg_path_selector_mode field in ARS_PROFILE defines the scope at which ARS selection for NH groups is applied.
+The system supports three distinct modes:
+
+**1. Global**:

```

Comment:
Is there such knob in SAI to enable ARS globally and avoid individual ARS interfaces/objects creation?